airflow/providers/amazon/aws/operators/appflow.py

```diff
@@ -0,0 +1,545 @@
+            response = self.hook.conn.describe_flow(flowName=self.name)
+
+        # Wait flow stops
+        self.log.info("Waiting flow run...")
+        while (
+            "mostRecentExecutionStatus" not in response["lastRunExecutionDetails"]
+            or response["lastRunExecutionDetails"]["mostRecentExecutionStatus"] == "InProgress"
+        ):
+            sleep(self.poll_interval)
+            response = self.hook.conn.describe_flow(flowName=self.name)
+
+        self.log.info("lastRunExecutionDetails: %s", response["lastRunExecutionDetails"])
+
+        if response["lastRunExecutionDetails"]["mostRecentExecutionStatus"] == "Error":
+            raise Exception(f"Flow error:\n{json.dumps(response, default=str)}")
+
+        return response["lastRunExecutionDetails"]["mostRecentExecutionStatus"]
+
+
+class AppflowRunOperator(AppflowBaseOperator):
+    """
+    Execute a Appflow run with filters as is.
+
+    .. seealso::
+        For more information on how to use this operator, take a look at the guide:
+        :ref:`howto/operator:AppflowRunOperator`
+
+    :param source: The source name (e.g. salesforce, zendesk)
+    :param name: The flow name
+    :param poll_interval: how often in seconds to check the query status
+    :param aws_conn_id: aws connection to use
+    :param region: aws region to use
+    """
+
+    def __init__(
+        self,
+        source: str,
+        name: str,
+        poll_interval: int = 20,
+        aws_conn_id: Optional[str] = "aws_default",
+        region: Optional[str] = None,
+        **kwargs,
+    ) -> None:
+        if source not in {"salesforce", "zendesk"}:
+            raise AirflowException(f"Source {source} is not supported for AppflowRunOperator!")
+        super().__init__(
+            source=source,
+            name=name,
+            flow_update=False,
+            source_field=None,
+            dt=None,
+            poll_interval=poll_interval,
+            aws_conn_id=aws_conn_id,
+            region=region,
+            **kwargs,
+        )
+
+
+class AppflowRunFullOperator(AppflowBaseOperator):
+    """
+    Execute a Appflow full run removing any filter.
+
+    .. seealso::
+        For more information on how to use this operator, take a look at the guide:
+        :ref:`howto/operator:AppflowRunFullOperator`
+
+    :param source: The source name (e.g. salesforce, zendesk)
+    :param name: The flow name
+    :param poll_interval: how often in seconds to check the query status
+    :param aws_conn_id: aws connection to use
+    :param region: aws region to use
+    """
+
+    def __init__(
+        self,
+        source: str,
+        name: str,
+        poll_interval: int = 20,
+        aws_conn_id: Optional[str] = "aws_default",
+        region: Optional[str] = None,
+        **kwargs,
+    ) -> None:
+        if source not in {"salesforce", "zendesk"}:
+            raise AirflowException(f"Source {source} is not supported for AppflowRunFullOperator!")
```

Comment:
Same comments re: using `SUPPORTED_SOURCES` and throwing a `ValueError` instead. WDYT?

Reply:
The exception class was already updated.

But the fact that the `SUPPORTED_SOURCES` is not spread through all operators is intentional.

Each operator must be tested and evolved independently, and the support by each source is specific to each operator.

So the `SUPPORTED_SOURCES ` global variable is just a concatenation of all possible sources until now across all operators, but it is not supposed to be used to validate each one of them individually.

I want to avoid the situation: Someone updates the global `SUPPORTED_SOURCES` to add support to a new source to operator A and unintentionally gives operator B support to that new source.

Does it make sense to you @josh-fell?